File: Java/18-github-copilot-app/src/main/java/com/githubcopilot/demo/WorkItem.java
package com.githubcopilot.demo;

import java.time.Instant;

public record WorkItem(
        String id,
        String title,
        Priority priority,
        Status status,
        int estimate,
        Instant createdAt,
        Instant dueAt) {

    public enum Priority {
        LOW,
        NORMAL,
        HIGH,
        CRITICAL
    }

    public enum Status {
        DRAFT,
        READY,
        IN_PROGRESS,
        DONE
    }
}

File: CSharp/18-github-copilot-app/src/DeliveryDashboard/WorkItem.cs
namespace DeliveryDashboard;

public enum WorkItemPriority
{
    Low,
    Normal,
    High,
    Critical
}

public enum WorkItemStatus
{
    Draft,
    Ready,
    InProgress,
    Done
}

public sealed record WorkItem(
    string Id,
    string Title,
    WorkItemPriority Priority,
    WorkItemStatus Status,
    int Estimate,
    DateTimeOffset CreatedAt,
    DateTimeOffset? DueAt);

public sealed record SprintPlan(IReadOnlyList<WorkItem> Items, int TotalEstimate);

public enum RiskLevel
{
    DueSoon,
    Overdue
}

public sealed record AtRiskItem(WorkItem Item, RiskLevel Risk);